donut/modules/auth/routes.py

```diff
@@ -12,13 +13,52 @@ def login():
 @blueprint.route('/login/submit', methods=['POST'])
 def login_submit():
     """Handle authentication."""
-    username = flask.request.form.get('username', None)
+    mask_and_user = flask.request.form.get('username', None)
     password = flask.request.form.get('password', None)
 
+    # Need to take care of the case when masquerading occurs.
+    username = None
+    mask = None
+
+    if mask_and_user is not None:
+        # Check for the character to split.
+        if ':' in mask_and_user:
+            # If it's there, we assume the second part is a mask name.
+            split = mask_and_user.split(':')
```

Comment:
Could just do the `split()` and then check how many elements are in the split

Reply:
I actually meant instead of the `':' in mask_and_user` check; if the split contains only 1 element, then it's just the username